File: fapi/src/main/java/com/iquiz/fapi/models/TestLiterature.java
package com.iquiz.fapi.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)

public class TestLiterature {
    private int idTest;
    private int idLiterature;
    private String testName;
    private String link;

    public TestLiterature() {
    }

    public TestLiterature(int idTest, int idLiterature, String testName, String link) {
        this.idTest = idTest;
        this.idLiterature = idLiterature;
        this.testName = testName;
        this.link = link;
    }

    public int getIdTest() {
        return idTest;
    }

    public void setIdTest(int idTest) {
        this.idTest = idTest;
    }

    public int getIdLiterature() {
        return idLiterature;
    }

    public void setIdLiterature(int idLiterature) {
        this.idLiterature = idLiterature;
    }

    public String getTestName() {
        return testName;
    }

    public void setTestName(String testName) {
        this.testName = testName;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }
}

File: frontend/src/app/modules/menu/components/materials/testLiterature.ts
export class TestLiterature{
  idTest: string;
  testName: string;
  idLiterature: string;
  link: string;

  static cloneBase(testLiterature: TestLiterature): TestLiterature {
    const clonedTestLiterature: TestLiterature = new TestLiterature();
    clonedTestLiterature.idTest = testLiterature.idTest;
    clonedTestLiterature.testName = testLiterature.testName;
    clonedTestLiterature.idLiterature = testLiterature.idLiterature;
    clonedTestLiterature.link = testLiterature.link;
    return clonedTestLiterature;
  }
}
